hardware_interface/include/hardware_interface/hardware_component_interface.hpp

```diff
@@ -40,13 +40,27 @@
 #include "rclcpp/logging.hpp"
 #include "rclcpp/node_interfaces/node_clock_interface.hpp"
 #include "rclcpp/time.hpp"
+#include "rclcpp/version.h"
 #include "rclcpp_lifecycle/node_interfaces/lifecycle_node_interface.hpp"
 #include "rclcpp_lifecycle/state.hpp"
 #include "realtime_tools/async_function_handler.hpp"
 
 namespace hardware_interface
 {
 
+static inline rclcpp::NodeOptions get_hardware_component_node_options()
+{
+  rclcpp::NodeOptions node_options;
+// \note The versions conditioning is added here to support the source-compatibility with Humble
+#if RCLCPP_VERSION_MAJOR >= 21
+  node_options.enable_logger_service(true);
+#else
+  node_options.allow_undeclared_parameters(true);
+  node_options.automatically_declare_parameters_from_overrides(true);
```

Comment:
```suggestion
```
do we need this?

Reply:
I'm not sure. I copied this from the controller node options. I think it is not needed. Why do you think?

Reply:
@saikishor You didn't answer @christophfroehlich  question about whether we need the following for older verisons:

```
node_options.allow_undeclared_parameters(true);
node_options.automatically_declare_parameters_from_overrides(true);
```